OpenRA.Mods.Common/Traits/Render/WithSwitchableOverlay.cs

```diff
@@ -0,0 +1,174 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	[Desc("Renders a decorative animation on units and buildings. Overlay switching controlled by " + nameof(PauseOnCondition) + ".")]
+	public class WithSwitchableOverlayInfo : PausableConditionalTraitInfo, IRenderActorPreviewSpritesInfo, Requires<RenderSpritesInfo>, Requires<BodyOrientationInfo>
+	{
+		[Desc("Image used for this decoration. Defaults to the actor's type.")]
+		public readonly string Image = null;
+
+		[FieldLoader.Require]
+		[SequenceReference(nameof(Image), allowNullImage: true)]
+		[Desc("Animation to play when the trait is enabling and disabling.")]
+		public readonly string SwitchingSequence = null;
+
+		[FieldLoader.Require]
+		[SequenceReference(nameof(Image), allowNullImage: true)]
+		[Desc("Animation to play when the trait is enabled")]
+		public readonly string EnabledSequence = null;
+
+		[FieldLoader.Require]
+		[SequenceReference(nameof(Image), allowNullImage: true)]
+		[Desc("Animation to play when the trait is disabled.")]
+		public readonly string DisabledSequence = null;
+
+		[Desc("Position relative to body")]
+		public readonly WVec Offset = WVec.Zero;
+
+		[PaletteReference(nameof(IsPlayerPalette))]
+		[Desc("Custom palette name")]
+		public readonly string Palette = null;
+
+		[Desc("Custom palette is a player palette BaseName")]
+		public readonly bool IsPlayerPalette = false;
+
+		public readonly bool IsDecoration = false;
+
+		[Desc("Duration in ticks when overlay switching.")]
+		public readonly int SwitchingLevel = 20;
```

Comment:
```suggestion
		[Desc("How long (in ticks) should the switching animation play?")]
		public readonly int SwitchingDuration = 20;
```

Reply:
I think we need consistency bettween "SwitchingLevelOnSpawn" and "SwitchingLevel".